### PR TITLE
Versions and publish

### PR DESCRIPTION
**Pull request by talef-perim81**
Added NX plugins for deploying and versioning with conventional commits.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,6 @@ testem.log
 # System Files
 .DS_Store
 Thumbs.db
+
+# Secrets
+.npmrc
```
Comment (talef-perim81) [inline .gitignore]: I added a `.npmrc` with a secret npm key for this project so we won't have to change the global npm key.

Comment (yurikrupnik) [reply .gitignore]: key should not be part of the project FYI.

File: apps/internal/jarvis/next.config.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const withNx = require('@nrwl/next/plugins/with-nx');
```
Comment (yurikrupnik) [inline apps/internal/jarvis/next.config.js:2]: tell me about this part plz.

File: apps/internal/jarvis/public/star.svg [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yurikrupnik) [inline apps/internal/jarvis/public/star.svg:1]: dont we have those svg's in s3 bucket?

File: libs/internal/jarvis-routes/src/lib/internal-jarvis-routes.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { Pagination } from '@tal-test/pagination';
```
Comment (yurikrupnik) [inline libs/internal/jarvis-routes/src/lib/internal-jarvis-routes.tsx]: this should be in a folder called internal, not internal in the name

File: apps/internal/jarvis/public/star.svg (hunk 3 above)
Comment (talef-perim81) [reply apps/internal/jarvis/public/star.svg:1]: This is auto generated by the framework, I was focusing the modules and versioning task and less about the files themselves.
Let me know if you want me to remove it.

File: apps/internal/jarvis/next.config.js (hunk 2 above)
Comment (talef-perim81) [reply apps/internal/jarvis/next.config.js:2]: This is auto generated by NX, it updates webpack config, I couldn't find any info about it online, but you can enter the `with-nx.js` file and see the comments explaining what does it do.

File: libs/internal/jarvis-routes/src/lib/internal-jarvis-routes.tsx (hunk 4 above)
Comment (talef-perim81) [reply libs/internal/jarvis-routes/src/lib/internal-jarvis-routes.tsx]: The file name was autogenerated, as said in other comments, I was focusing something else, let me know if you want me to change it to another name.

**Pull request merged**